spec/lib/search_spec.rb

```diff
@@ -2002,6 +2002,16 @@ def search_with_bookmarks
       expect(Search.execute("l sam").posts.map(&:id)).to eq([post2.id, post1.id])
     end
 
+    it "can find by oldest" do
+      topic1 = Fabricate(:topic, title: "I do not like that Sam I am")
+      post1 = Fabricate(:post, topic: topic1, raw: "sam is a sam sam sam") # score higher
+
+      topic2 = Fabricate(:topic, title: "I do not like that Sam I am 2", created_at: 5.minutes.ago)
+      post2 = Fabricate(:post, topic: topic2, created_at: 5.minutes.ago)
+
+      expect(Search.execute("sam ORDER:oldest").posts.map(&:id)).to eq([post2.id, post1.id])
```

Comment:
```suggestion
      expect(Search.execute("sam").posts.map(&:id)).to eq([post1.id, post2.id])
      expect(Search.execute("sam ORDER:oldest").posts.map(&:id)).to eq([post2.id, post1.id])
```

I think having additional assertions to ensure that the new `order:oldest` filter returns results that is different from the default is important. Otherwise, we're always assuming that the fabrication is always correct which may not always be the case.